### ISSUE TITLE
resource/awscc_ec2_ipam - Attributes Set operating_regions unable to be updated

### ISSUE THREAD

**Issue by drewmullen** [hashicorp/terraform-provider-awscc#321]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the Cloudformation Open Coverage Roadmap.

### Affected Resource(s)

* awscc_ec2_ipam

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "awscc_ec2_ipam" "main" {
    description = "AWS IP Address Manager"
    operating_regions = [
           {
               region_name = "us-east-1"
            },
           {
               region_name = "us-west-1"
            },
    ]
    tags = [{
        key = "test"
        value = "test"
    }]
}
```

Updated:
```terraform
resource "awscc_ec2_ipam" "main" {
    description = "AWS IP Address Manager"
    operating_regions = [
           {
               region_name = "us-east-1"
            },
           {
               region_name = "us-west-1"
            },
         {
               region_name = "us-east-2"
            },
    ]
    tags = [{
        key = "test"
        value = "test"
    }]
}
```

### Debug Output

Can provide upon request

### Expected Behavior

should show update required to aws_ec2_ipam to add new regions

### Actual Behavior

```
$ tf apply
awscc_ec2_ipam.main: Refreshing state... [id=ipam-084156d5196fa2c20]
╷
│ Error: State Read Error
│ 
│   with awscc_ec2_ipam.main,
│   on main.tf line 12, in resource "awscc_ec2_ipam" "main":
│   12:     operating_regions = [
│   13:            {
│   14:                region_name = "us-east-1"
│   15:             },
│   16:            {
│   17:                region_name = "us-west-1"
│   18:             },
│   19:          {
│   20:                region_name = "us-east-2"
│   21:             },
│   22:     ]
│ 
│ An unexpected error was encountered trying to read an attribute from the state. This is always an error in the provider. Please report the following to the provider developer:
│ 
│ ElementKeyValue(tftypes.Object["region_name":tftypes.String]<"region_name":tftypes.String<"us-east-2">>).AttributeName("region_name") still remains in the path: step cannot be applied to
│ this value
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply` with updated `operating_regions` attribute

### References

* #0000

**Comment by drewmullen** [comment]
I think this is at the attribute set refresh level - not specific to a resource. i got the same error with `awscc_ec2_ipam_pool`

```
│ Error: State Read Error
│ 
│   with awscc_ec2_ipam_pool.ipv4_private_default,
│   on main.tf line 76, in resource "awscc_ec2_ipam_pool" "ipv4_private_default":
│   76:     provisioned_cidrs = [{
│   77:         cidr = "172.2.0.0/16"
│   78:     }]
│ 
│ An unexpected error was encountered trying to read an attribute from the state. This is always an error in the provider. Please report the following to the provider developer:
│ 
│ ElementKeyValue(tftypes.Object["cidr":tftypes.String]<"cidr":tftypes.String<"172.2.0.0/16">>).AttributeName("cidr") still remains in the path: step cannot be applied to this value
```

**Comment by ewbankkit** [comment]
@drewmullen Thanks for raising this issue.
This has been addressed via https://github.com/hashicorp/terraform-plugin-framework/pull/185, which is included in #306.

**Comment by ewbankkit** [comment]
Hmm, this is because I am running in `us-west-2` and the initial `terraform apply` which didn't explicitly list `us-west-2` as an operating region implicitly adds it - hence `terraform plan` showing that `us-west-2` is to be removed.

**Comment by drewmullen** [comment]
yeah - in the aws provider i got around that by requiring users to input the current region lol

IMO that is acceptable to explain in documentation

**Comment by drewmullen** [comment]
this was resolved with [0.9](https://github.com/hashicorp/terraform-provider-awscc/releases/tag/v0.9.0)! thanks @ewbankkit !
